lib/DxcSupport/FileIOHelper.cpp

```diff
@@ -870,25 +935,28 @@ HRESULT DxcGetBlobAsUtf8(IDxcBlob *pBlob, IMalloc *pMalloc, IDxcBlobUtf8 **pBlob
   // Reuse or copy the underlying blob depending on null-termination
   if (codePage == CP_UTF8) {
     utf8CharCount = blobLen;
-    if (IsBufferNullTerminated(pBlob->GetBufferPointer(), blobLen, CP_UTF8)) {
+    if (IsBufferNullTerminated(bufferPointer, blobLen, CP_UTF8)) {
       // Already null-terminated, reference other blob's memory
       InternalDxcBlobUtf8* internalEncoding;
       hr = InternalDxcBlobUtf8::CreateFromBlob(pBlob, pMalloc, true, CP_UTF8, &internalEncoding);
       if (SUCCEEDED(hr)) {
+        // Adjust if buffer has BOM; blobLen is already adjusted.
+        if (bomSize)
+          internalEncoding->AdjustPtrAndSize(bomSize, blobLen);
```

Comment:
Is it possible that there be a BOM when the codepage is already determined by `pSourceBlob->GetEncoding`? If so, we may still need to make this adjustment

Reply:
A BlobEncoding created with encoding known and codePage != CP_ACP will have had the BOM stripped in `DxcCreateBlob`.  See:
```
  unsigned bomSize = (encodingKnown && codePage != CP_ACP)
                         ? GetBomLengthFromBytes((const char *)pPtr, size)
                         : 0;
  if (bomSize) {
    // Adjust pPtr and size to skip BOM.
    ...
```